js/oceanex.js

```diff
@@ -0,0 +1,617 @@
+        const data = this.safeValue (response, 'data');
+        return this.parseTrades (data, market, since, limit);
+    }
+
+    parseTrade (trade, market = undefined) {
+        let side = this.safeValue (trade, 'side');
+        if (side === 'bid') {
+            side = 'buy';
+        } else if (side === 'ask') {
+            side = 'sell';
+        }
+        let symbol = undefined;
+        let marketId = this.safeValue (trade, 'market');
+        if (marketId !== undefined) {
+            marketId = marketId.toUpperCase ();
+            if (marketId in this.markets_by_id) {
+                market = this.markets_by_id[marketId];
+                symbol = market['symbol'];
+            } else {
+                symbol = marketId;
+            }
+        }
+        if (symbol === undefined) {
+            if (market !== undefined) {
+                symbol = market['symbol'];
+            }
+        }
+        let timestamp = this.safeInteger (trade, 'created_on');
+        if (timestamp === undefined) {
+            timestamp = this.parse8601 (this.safeString (trade, 'created_at'));
+        } else {
+            timestamp = timestamp * 1000;
+            // Get milliseconds
+        }
+        return {
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'id': this.safeString (trade, 'id'),
+            'order': undefined,
+            'type': 'limit',
+            'takerOrMaker': undefined,
+            'side': side,
+            'price': this.safeFloat (trade, 'price'),
+            'amount': this.safeFloat (trade, 'volume'),
+            'cost': undefined,
+            'fee': undefined,
+        };
+    }
+
+    async fetchTime (params = {}) {
+        const response = await this.publicGetTimestamp ();
+        const data = this.safeValue (response, 'data');
+        return data;
+    }
+
+    async fetchTradingFees (params = {}) {
+        const response = await this.publicGetFeesTrading ();
+        const data = this.safeValue (response, 'data');
+        let result = [];
+        for (let i = 0; i < data.length; i++) {
+            let group = data[i];
+            let maker = this.safeValue (group, 'ask_fee');
+            let taker = this.safeValue (group, 'bid_fee');
+            result.push ({
+                'info': group,
+                'maker': this.safeFloat (maker, 'value'),
+                'taker': this.safeFloat (taker, 'value'),
+            });
+        }
+        return result;
+    }
+
+    async fetchKey (params = {}) {
+        const response = await this.privateGetKey (this.extend (params));
+        const data = this.safeValue (response, 'data');
+        return data;
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        const response = await this.privateGetMembersMe (this.extend (params));
```

Comment:
We need to remove `this.extend`, because it is to extend one dictionary with another. There's just one dictionary `params` here, therefore `this.extend` is not required, and `params` is required.

Same thing everywhere (`fetchKey ()`) and maybe other places in this file.